source/s3_endpoint_resolver/s3_endpoint_resolver.c

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/s3/s3_endpoint_resolver.h>
+#include <aws/sdkutils/endpoints_rule_engine.h>
+#include <aws/sdkutils/partitions.h>
+
+struct aws_endpoints_rule_engine *aws_s3_endpoint_resolver_new(struct aws_allocator *allocator) {
+    struct aws_endpoints_ruleset *ruleset = NULL;
+    struct aws_partitions_config *partitions = NULL;
+    struct aws_endpoints_rule_engine *rule_engine = NULL;
+
+    struct aws_byte_cursor ruleset_cursor = aws_byte_cursor_from_c_str(aws_s3_endpoint_rule_set);
```

Comment:
utterly trivial: from_c_str() means an O(N) search to the end of the c-string

you could make these extern variables `aws_string` and define them using `AWS_STATIC_STRING_FROM_LITERAL()` or... who cares this is a call once thing

Reply:
Thanks, fixed by generating a byte cursor directly.